src/ott/problems/linear/barycenter_problem.py

```diff
@@ -176,6 +176,56 @@ def tree_flatten(self) -> Tuple[Sequence[Any], Dict[str, Any]]:  # noqa: D102
   @classmethod
   def tree_unflatten(  # noqa: D102
       cls, aux_data: Dict[str, Any], children: Sequence[Any]
-  ) -> "BarycenterProblem":
+  ) -> "FreeBarycenterProblem":
     y, b, weights = children
     return cls(y=y, b=b, weights=weights, **aux_data)
+
+
+@jax.tree_util.register_pytree_node_class
+class FixedBarycenterProblem:
+  """Fixed-support Wasserstein barycenter problem.
+
+  Args:
+    geom: :class:`~ott.geometry.geometry.Geometry` object.
+    a: batch of histograms of shape ``[batch, num_a]`` where `num_a` matches
+      the first value of `geom.shape`.
+    weights: ``[batch,]`` positive weights summing to 1. Uniform by default.
```

Comment:
Maybe :math\`1\`? If yes, would also add it in the docstring of ``weights``.

Reply:
thanks!